filters/SampleFilter.cpp

```diff
@@ -90,8 +103,24 @@ PointViewSet SampleFilter::run(PointViewPtr view)
     PointViewPtr output = view->makeNew();
     for (PointRef point : *view)
     {
-        if (voxelize(point))
+        bool kept = voxelize(point);
+        if (kept)
+        {
+            // Write a 1 if the user set a 'dimension' argument
+            if (m_dimension != pdal::Dimension::Id::Unknown)
+                point.setField(m_dimension, 1);
+
             output->appendPoint(*view, point.pointId());
+        } else
+        {
+            // If the user set a dimension, write 0s be cause we didn't
+            // keep the point. Otherwise we drop it on the floor.
+            if (m_dimension != pdal::Dimension::Id::Unknown)
+            {
+                point.setField(m_dimension, 0);
+                output->appendPoint(*view, point.pointId());
```

Comment:
`appendPoint` is incorrect here.

===

You could use this litter helper or move its logic into `voxelize`

```
bool func(point)
{
  bool keep = voxelize(point);
  if (m_dimension != pdal::Dimension::Id::Unknown)
  {
    point.setField(m_dimension, keep);
    keep = true;
  }
  return keep;
}
```
In run():

```
if (func(point))
  output->appendPoint(*view, point.pointId());
``` 

In processOne():

```
return func(point);
```

Reply:
updated as suggested